www/templates/intro/intro.html

```diff
@@ -13,7 +13,7 @@
         <ng-include src="'templates/intro/saveTokenFile.html'"></ng-include>
       </ion-slide>
       <ion-slide>
-        <ng-include ng-if="currentToken" src="surveyFile"></ng-include>
+        <ng-include ng-if="currentToken && slideIndex>3" src="surveyFile"></ng-include>
```

Comment:
some historical context on this, just to potentially think of a less hacky solution and to keep in mind while redesigning.

I structured it this way because the existing demographic survey can take a while to load from the server, and I didn't want people to wait around while it loaded. We cannot load it until all the consent is complete, because we do not contact the server until then. So I added the "saveToken" screen to distract the user while loading the survey in the background. You should be able to read the blame around the survey screen for details.

> I am guessing that when you move to the next slide of an ion-slide, it looks ahead to the next slide and creates it ahead of time.

However, IIRC, all `ion-slides` are created at the beginning. So by default, we would try to load the survey without configuring a token, which would fail. So I added the `ng-if`. The idea is that we set `$scope.currentToken` when the login is done. So then this screen loads and hopefully the survey is loaded by the time the user clicks past the save token screen.

What appears to have changed now is that the survey slide is loaded in front of the save token slide. I can actually see the save token slide (with part of the token) on the side. I don't think this was happening before, even during the react migration. Not sure what changed.

Reply:
I think what changed is the Enketo version. I can see that the issue only happens when the Enketo Form object is initialized. When I commented out the form initialization, the survey screen did not overtake the OPCode screen.

I think there is some code in the Enketo initialization that forces the survey into view if it is not in view.

I think this has been happening for a long time, but not resolved until now because it only happens in this specific scenario: 1) the existing survey request finishes before you get to the survey page and 2) no existing survey was found

Reply:
If this hack is ok for now, I am pretty sure we won't have the problem when we aren't using ion-slides anymore.
I want to redesign the intro pretty significantly.